lib/redmine_goodies_macros.rb

```diff
@@ -0,0 +1,81 @@
+module RedmineGoodies
+    module Macros
+        # Retrieves a mapping of note references (that contain questions using the question macro) to their corresponding answers.
+        def self.get_all_answers(issue_id)
+            issue = Issue.find_by(id: issue_id)    
+            journals = issue.journals.order(:created_on)
+
+            # eg. {"#note-19"=>[{:question=>"1", :response_from=>"#note-20"}, {:question=>"2", :response_from=>"#note-20"}
+            answered_questions = {}
+
+            # eg. "{{answer(#note-56, 1/}}", the regex below will extract:
+            # '#note-56' and '1'
+            regex = /\{\{answer\((#\w+-\d+),\s*(\d+)\/\)\}\}/
+
+            # Iterate through each journal and its notes to search for answers to questions
+            journals.each do |journal|
+                current_notes = journal.notes
+                next if current_notes.blank?
+                response_from_note_id = "#note-#{journals.index(journal) + 1}"  # getting note_number (not equals with journal.id)
+                
+                # search in notes for answers to questions
+                matches = current_notes.scan(regex)
+                matches.each do |match|
+                    note_ref = match[0]  # the note reference from which the questions come
+                    question_number = match[1]  # the answered question number
+                    
+                    answered_questions[note_ref] ||= []
+                    unless answered_questions[note_ref].any? { |answer| answer[:question] == question_number && answer[:response_from] == response_from_note_id }
+                        answered_questions[note_ref] << { question: question_number, response_from: response_from_note_id }
+                    end
+                end
+            end
+            answered_questions
+        end
+
+        Redmine::WikiFormatting::Macros.register do
+            desc "Custom macro for rendering questions"
+            macro :questions do |obj, args, text|
```

Comment:
I'm wondering if it is possible to restrict a macro to only issues. I think it is also available on wiki, correct?

Reply:
Yes, they are also available on the wiki, but I added `if obj.is_a?(Journal)`, which restricts the usage to Issues only.